### ISSUE TITLE
`eth_getLogs` contention can cause a node to revert to syncing

### ISSUE THREAD

**Issue by ryanschneider** [ethereum/go-ethereum#17611]
#### System information

Geth version: `1.8.13` (but present in .14 and .15 too)
OS & Version: Linux

#### Expected behaviour

The scheduling used inside geth allows multiple getLogs queries to be run concurrently without affecting devp2p performance.

#### Actual behaviour

As mentioned in #17610, when running 10s of concurrent getLogs queries on a node, the scheduling contention gets to the point where importing new blocks will fall behind, and devp2p connections will start timing out:

```
Sep 07 22:25:35 ip-10-0-24-237 geth[30308]: INFO [09-07|22:25:35.808] Imported new chain segment               blocks=1   txs=66   mgas=3.694   elapsed=1.599s    mgasps=2.309  number=6290691 hash=d461b8…aa70c4 cache=93.05mB
Sep 07 22:25:48 ip-10-0-24-237 geth[30308]: INFO [09-07|22:25:48.606] Imported new chain segment               blocks=1   txs=55   mgas=7.989   elapsed=1.627s    mgasps=4.909  number=6290692 hash=692938…1cd22b cache=93.72mB
Sep 07 22:26:30 ip-10-0-24-237 geth[30308]: WARN [09-07|22:26:30.398] Synchronisation failed, retrying         err="block download canceled (requested)"
```

#### Steps to reproduce the behaviour

Run tens of wide concurrent requests against a node concurrently.  I can provide examples if necessary.

I also have a *very* rough branch with some attempts at adding rate-limiting here:

https://github.com/ryanschneider/go-ethereum/tree/rate-limit-getLogs

Basically, by adding any sort of throttling to `Filter.checkMatches` I was able to prevent the `Synchronisation failed` messages from occurring.  I'm not convinced that's the actual best place to do any rate-listing though, it was just how I was exploring solutions.
